libc/utils/docgen/docgen.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python
+#
+# ====- Generate documentation for libc functions  ------------*- python -*--==#
+#
+# Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# ==-------------------------------------------------------------------------==#
+from argparse import ArgumentParser
+from pathlib import Path
+import sys
+import yaml
+
+
+def load_api():
+    p = Path(Path(__file__).resolve().parent, "api.yml")
+    api = p.read_text(encoding="utf-8")
+    return yaml.load(api, Loader=yaml.FullLoader)
+
+
+# TODO: we may need to get more sophisticated for less generic implementations.
+# Does libc/src/{hname minus .h suffix}/{fname}.cpp exist?
+def is_implemented(hname, fname):
```

Comment:
nit: Do you want to do type annotation for our python scripts?

Reply:
Yes, I think it's a nice touch. I don't have much experience writing/checking these, but I have seen them before.

IIRC, there's a separate utility for checking these than the python vs/runtime itself?  Got any pointers?

cc @JustinStitt

Reply:
done in https://github.com/llvm/llvm-project/pull/87682/commits/aba909db2c61a8ab2e2b63309c749b98ca055179

Reply:
The last time I checked, type hints were ignored by CPython/the Python interpreter, and just used by language servers.

Reply:
Or some linter tools to help enforcing them.

Reply:
@nickdesaulniers stuff like pylance, [ruff](https://github.com/astral-sh/ruff) or [basedpyright](https://github.com/DetachHead/basedpyright) can parse these type hints. 

count me as a +1 for type-hinting everywhere!